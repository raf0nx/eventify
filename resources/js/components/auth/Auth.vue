<template>
	<v-main>
		<v-container class="fill-height py-0" fluid>
			<v-row class="fill-height">
				<v-col cols="12" md="7" class="pa-0 auth-panel">
					<div
						class="auth-panel__container pa-4 d-flex flex-column justify-space-between"
					>
						<div class="d-flex logo">
							<v-icon class="white--text mr-1">mdi-buffer</v-icon>
							<span class="logo__text white--text font-weight-bold"
								>Eventify</span
							>
						</div>
						<div class="d-flex flex-column align-center mb-16">
							<h2 class="auth-panel__title white--text">
								Welcome back!
							</h2>
							<span
								class="auth-panel__subtitle grey--text text--lighten-3 mt-4"
								>Sign In to continue using my application!</span
							>
						</div>
						<div class="d-flex justify-center">
							<span class="white--text font-weight-bold"
								>v. 0.0</span
							>
						</div>
					</div>
				</v-col>
				<transition name="scale" mode="out-in">
					<router-view />
				</transition>
			</v-row>
		</v-container>
	</v-main>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	@Component
	export default class Auth extends Vue {}
</script>

<style>
	.auth-panel {
		background-image: url("https://i.pinimg.com/originals/c4/6f/3c/c46f3cfbff1b198e0276e4d577a4464c.png");
		background-position: center;
		background-repeat: no-repeat;
		background-size: cover;
	}

	.auth-panel__container {
		height: 100%;
	}

	.auth-panel__title {
		font-size: 4rem;
	}

	.auth-panel__subtitle {
		font-size: 1.5rem;
	}

	.logo__text {
		font-size: 1.5rem;
	}

	.auth-form__title {
		font-size: 3rem;
	}
</style>