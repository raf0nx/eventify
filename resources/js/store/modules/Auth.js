import router from "../../router/router";
import AuthService from "../../services/AuthService";

export const namespaced = true;

export const state = {
    user: null,
};

export const mutations = {
    SET_USER(state, user) {
        state.user = user;
    },
};

export const actions = {
    async getAuthUser({ commit }) {
        try {
            const authUser = await AuthService.getAuthUser();
            commit("SET_USER", authUser.data);
            return authUser.data;
        } catch (error) {
            commit("SET_USER", null);
        }
    },
    logout({ commit }) {
        try {
            AuthService.logout();
            commit("SET_USER", null);
            if (router.currentRoute.name !== "Login") {
                router.push({ name: "Login" });
            }
        } catch (error) {
            console.log(error);
        }
    }
};

export const getters = {
    authUser: state => {
        return state.user;
    },
};
