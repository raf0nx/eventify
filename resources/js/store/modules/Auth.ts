import {
    Module,
    VuexModule,
    Mutation,
    Action,
    getModule
} from "vuex-module-decorators";

import store from "@/store/store.ts";
import router from "@/routes/router.ts";
import * as AuthService from "@/services/AuthService.ts";

export interface AuthState {
    user: any;
}

@Module({ dynamic: true, store, name: "auth" })
class Auth extends VuexModule implements AuthState {
    public user: any = null;

    @Mutation
    private SET_USER(userPayload: any) {
        this.user = userPayload;
    }
    @Action
    public async getAuthUser() {
        try {
            const authUser = await AuthService.getAuthUser();
            this.SET_USER(authUser.data);
            return authUser.data;
        } catch (error) {
            this.SET_USER(null);
        }
    }

    @Action
    public async logout() {
        try {
            await AuthService.logout();
            this.SET_USER(null);
            if (router.currentRoute.name !== "Login") {
                router.push({ name: "Login" });
            }
        } catch (error) {
            console.log(error);
        }
    }

    public get authUser() {
        return this.user;
    }
}

export const AuthModule = getModule(Auth);
