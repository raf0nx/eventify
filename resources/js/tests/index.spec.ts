import Auth from "@components/auth/Auth.vue";
import Vuetify from "vuetify";
import VueRouter from "vue-router";

import { createLocalVue, mount } from "@vue/test-utils";

describe("Auth.vue", () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
    });

    it("should work", () => {
        const wrapper = mount(Auth, { localVue, vuetify, router });
        expect(wrapper.html()).toMatchSnapshot();
    });
});