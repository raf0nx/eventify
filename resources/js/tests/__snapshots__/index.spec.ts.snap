// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth.vue should work 1`] = `
"<main class=\\"v-main\\" style=\\"padding: 0px 0px 0px 0px;\\">
  <div class=\\"v-main__wrap\\">
    <div class=\\"container fill-height py-0 container--fluid\\">
      <div class=\\"row fill-height\\">
        <div class=\\"pa-0 left-panel col-md-7 col-12\\">
          <div class=\\"left-panel-container pa-4 d-flex flex-column justify-space-between\\">
            <div class=\\"d-flex\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate white--text mr-1 mdi mdi-buffer theme--light\\"></i> <span class=\\"logo-text white--text font-weight-bold\\">Eventify</span></div>
            <div class=\\"d-flex flex-column align-center mb-16\\">
              <h2 class=\\"left-panel-title white--text\\">
                Welcome back!
              </h2> <span class=\\"left-panel-subtitle grey--text text--lighten-3 mt-4\\">Sign In to continue using my application!</span>
            </div>
            <div class=\\"d-flex justify-center\\"><span class=\\"white--text font-weight-bold\\">v. 0.0</span></div>
          </div>
        </div>
        <transition-stub name=\\"scale\\" mode=\\"out-in\\">
          <!---->
        </transition-stub>
      </div>
    </div>
  </div>
</main>"
`;
